src/main/java/com/denizenscript/clientizen/objects/properties/material/internal/MaterialEnumProperty.java

```diff
@@ -7,40 +7,58 @@
 import net.minecraft.state.property.EnumProperty;
 import net.minecraft.util.StringIdentifiable;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
+import java.util.stream.Collectors;
 
 @SuppressWarnings({"rawtypes", "unchecked"})
 public abstract class MaterialEnumProperty extends MaterialMinecraftProperty {
 
-    public static final Map<EnumProperty<?>, Conversion<?, ?>> conversions = new HashMap<>();
+    public static final Map<EnumProperty<?>, Conversion<?, ?>> CONVERSIONS = new HashMap<>();
 
-    record Conversion<IT extends Enum<IT> & StringIdentifiable, ET extends Enum<ET> & StringIdentifiable>(Class<ET> externalType, ET[] externalTypeConstants, IT[] internalTypeConstants,
-                                                                                                          IT toRemove) {}
+    public static class Conversion<IT extends Enum<IT> & StringIdentifiable, ET extends Enum<ET> & StringIdentifiable> {
+        // Entire enum conversion
+        Class<ET> externalType;
+        ET[] externalTypeConstants;
+        IT[] internalTypeConstants;
+        // Value removal
+        Set<String> toRemove;
+
+        public boolean shouldRemove(String value) {
+            return toRemove != null && toRemove.contains(value);
+        }
+    }
 
     public static <IT extends Enum<IT> & StringIdentifiable, ET extends Enum<ET> & StringIdentifiable> void convertEnum(EnumProperty<IT> internalProperty, Class<ET> externalType) {
-        conversions.put(internalProperty, new Conversion<>(externalType, externalType.getEnumConstants(), internalProperty.getType().getEnumConstants(), null));
+        Conversion conversion = CONVERSIONS.computeIfAbsent(internalProperty, k -> new Conversion<IT, ET>());
+        conversion.externalType = externalType;
+        conversion.externalTypeConstants = externalType.getEnumConstants();
+        conversion.internalTypeConstants = internalProperty.getType().getEnumConstants();
     }
 
-    public static <IT extends Enum<IT> & StringIdentifiable> void removeSingleValue(EnumProperty<IT> internalProperty, IT toRemove) {
-        conversions.put(internalProperty, new Conversion<>(null, null, null, toRemove));
+    public static <IT extends Enum<IT> & StringIdentifiable> void removeValues(EnumProperty<IT> internalProperty, IT... toRemove) {
+        CONVERSIONS.computeIfAbsent(internalProperty, k -> new Conversion()).toRemove = Arrays.stream(toRemove).map(StringIdentifiable::asString).collect(Collectors.toCollection(HashSet::new));
```

Comment:
`Collectors.toSet()`

Reply:
In this case it's specifically a hash set, as iirc `toSet` doesn't promise a specific type of set being returned